src/openrct2/world/Climate.cpp

```diff
@@ -311,8 +311,8 @@ static void ClimateDetermineFutureWeather(uint32_t randomValue)
     // Generate a random index with values 0 up to randomBias-1
     // and choose weather from the distribution table accordingly
     const auto& pattern = kClimatePatterns[EnumValue(gameState.Climate)][month];
-    size_t randomIndex = ((randomValue % 256) * pattern.randomBias) / 256;
-    auto nextWeather = pattern.distribution[randomIndex];
+    const auto randomIndex = ((randomValue % 256) * pattern.randomBias) / 256;
+    const auto nextWeather = pattern.distribution[randomIndex];
```

Comment:
Could you elaborate why you changed this back to `auto`? I get the added `const`ness, but was under the impression it needed to be made unsigned, explicitly.

Reply:
I was wondering that as well, considering line +314 using `auto` caused the corruption.

Also, maybe we should add an `assert` here to make sure it’s actually a valid value?

Reply:
Following should illustrate how the problem happened.
```cpp
uint32_t randValue = 0xF1F2F3F4;       // 11110001111100101111001111110100, sign bit set.
int32_t inputValue = randValue;        // Becomes negative due to the sign bit, this is where the issue started, the compiler did not warn about passing uint32_t as int32_t to the function.
int32_t moduloRes = inputValue % 256;  // Remains negative.
int32_t biasRes = moduloRes * bias;    // Bias is quite small so good chance it remains negative.
int32_t final = biasRes / 256;         // If negative it will remain negative.
```